src/App.tsx

```diff
@@ -1,44 +1,51 @@
-import React, { useEffect } from "react";
-import { useGetLaunchesByNameQuery } from "./store/launches/launches.api";
+import { useEffect, useState } from "react";
+import { useGetLaunchesByNameQuery } from "./api/launches.api";
 import { useDispatch, useSelector } from "react-redux";
-import {
-  addLanches,
-  setPage,
-  setTotalPage,
-} from "./store/launches/launchesSlice";
+import { addLaunches, setTotalPage } from "./store/launches/launchesSlice";
 import { RootState } from "./store";
 import LaunchesView from "./components/LaunchesView";
 
 function App() {
-  const page = useSelector((state: RootState) => state.launches.page);
   const totalPages = useSelector(
     (state: RootState) => state.launches.totalPages
   );
-
-  const { data, error, isLoading } = useGetLaunchesByNameQuery(page);
+  const [page, setPage] = useState(1);
+  const { data, isLoading } = useGetLaunchesByNameQuery(page);
   const dispatch = useDispatch();
 
   useEffect(() => {
     if (data) {
-      dispatch(addLanches(data.docs));
+      dispatch(addLaunches(data.docs));
       dispatch(setTotalPage(data.totalPages));
-      dispatch(setPage(data.page));
+      setPage(data.page);
     }
-  }, [data]);
+  }, [data, dispatch]);
 
-  const handleLoadMore = () => {
-    const nextPage = page + 1;
-    if (nextPage > totalPages) {
-      return;
+  const scrollHandler = (e: any) => {
```

Comment:
investigate solution to avoid "any"

Reply:
`useEffect(() => {
    const onScroll = e => {
      setScrollTop(e.target.documentElement.scrollTop);
    };
    window.addEventListener('scroll', onScroll);

    return () => window.removeEventListener('scroll', onScroll);
  }, [scrollTop]);

  useEffect(() => {
    const windowHeight = Math.max(
      document.body.scrollHeight,
      document.body.offsetHeight,
      document.documentElement.scrollHeight,
      document.documentElement.offsetHeight
    );
    if (scrollTop !== 0
      &&
      scrollTop + window.innerHeight > windowHeight - window.innerHeight
      &&
      status !== 'pending'
      &&
      pageNumber < totalPages
    ) {
      setPageNumber(pageNumber + 1);
      setScrollTop(0);
    }
  }, [scrollTop]);`

Reply:
`useEffect(() => {
    const onScroll = e => {
      setScrollTop(e.target.documentElement.scrollTop);
    };
    window.addEventListener('scroll', onScroll);

    return () => window.removeEventListener('scroll', onScroll);
  }, [scrollTop]);

  useEffect(() => {
    const windowHeight = Math.max(
      document.body.scrollHeight,
      document.body.offsetHeight,
      document.documentElement.scrollHeight,
      document.documentElement.offsetHeight
    );
    if (scrollTop !== 0
      &&
      scrollTop + window.innerHeight > windowHeight - window.innerHeight
      &&
      status !== 'pending'
      &&
      pageNumber < totalPages
    ) {
      setPageNumber(pageNumber + 1);
      setScrollTop(0);
    }
  }, [scrollTop]);`